must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -233,4 +252,45 @@ protected boolean isSubtypeWithElements(
       return superVal.containsAll(subVal);
     }
   }
+
+  private class MustCallTreeAnnotator extends TreeAnnotator {
+    public MustCallTreeAnnotator(MustCallAnnotatedTypeFactory mustCallAnnotatedTypeFactory) {
+      super(mustCallAnnotatedTypeFactory);
+    }
+
+    // When they appear in the body of a method or constructor, treat non-owning parameters
+    // as bottom regardless of their declared type.
+    @Override
+    public Void visitIdentifier(IdentifierTree node, AnnotatedTypeMirror type) {
+      Element elt = TreeUtils.elementFromTree(node);
+      /// This code derived from ElementUtils#enclosingClass
+      Element enclosing = elt;
```

Comment:
Does it work to check if `elt.getKind() == ElementKind.PARAMETER` to see if the identifier is a parameter?

Reply:
This is a good idea, and it does help with my concerns about scoping. I added a check at the beginning of the method that returns early if the identifier isn't a parameter of the method.

Even with that change we can't remove anything from the rest of the method, though. We need to find the formal parameter that has the same name as the element, because that's the only place we can check for an `@Owning` annotation.

Reply:
So calling `getDeclAnnotation(elt, Owning.class)` doesn't work?  That's surprising to me.  I would think that `elt` _is_ the element for the formal parameter.

Reply:
You're right, it is. I must not have had enough caffeine yet today. The code is now much, much simpler - thanks!